pkg/webui/components/input/index.js

```diff
@@ -39,103 +38,47 @@ const m = defineMessages({
-      'new-password',
-      'off',
-      'on',
-      'url',
-      'username',
-    ]),
-    children: PropTypes.node,
-    className: PropTypes.string,
-    code: PropTypes.bool,
-    component: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-    disabled: PropTypes.bool,
-    error: PropTypes.bool,
-    forwardedRef: PropTypes.shape({ current: PropTypes.shape({}) }),
-    icon: PropTypes.string,
-    inputRef: PropTypes.shape({ current: PropTypes.shape({}) }),
-    inputWidth: PropTypes.inputWidth,
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func,
-    }).isRequired,
-    label: PropTypes.string,
-    loading: PropTypes.bool,
-    max: PropTypes.number,
-    onBlur: PropTypes.func,
-    onChange: PropTypes.func,
-    onEnter: PropTypes.func,
-    onFocus: PropTypes.func,
-    placeholder: PropTypes.message,
-    readOnly: PropTypes.bool,
-    sensitive: PropTypes.bool,
-    showPerChar: PropTypes.bool,
-    title: PropTypes.message,
-    type: PropTypes.string,
-    valid: PropTypes.bool,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    warning: PropTypes.bool,
-  }
-
-  static defaultProps = {
-    action: undefined,
-    actionDisable: false,
-    append: null,
-    autoComplete: 'off',
-    children: undefined,
-    className: undefined,
-    code: false,
-    component: 'input',
-    disabled: false,
-    error: false,
-    /** Default `inputWidth` value is set programmatically based on input type. */
-    inputWidth: undefined,
-    icon: undefined,
-    label: undefined,
-    loading: false,
-    max: undefined,
-    onFocus: () => null,
-    onBlur: () => null,
-    onChange: () => null,
-    onEnter: () => null,
-    placeholder: undefined,
-    readOnly: false,
-    sensitive: false,
-    showPerChar: false,
-    title: undefined,
-    type: 'text',
-    valid: false,
-    value: '',
-    warning: false,
-    inputRef: null,
-    forwardedRef: null,
-  }
-
-  constructor(props) {
-    super(props)
-
-    this.state = {
-      focus: false,
-      hidden: props.sensitive,
-    }
-  }
-
-  input = React.createRef(null)
-
-  _computeByteInputWidth() {
-    const { showPerChar, max, sensitive } = this.props
+const Input = React.forwardRef((props, ref) => {
+  const {
+    action,
+    actionDisable,
+    append,
+    autoComplete,
+    children,
+    className,
+    code,
+    component,
+    disabled,
+    error,
+    forwardedRef,
+    icon,
+    inputRef,
+    inputWidth,
+    label,
+    loading,
+    onBlur,
+    onChange,
+    onEnter,
+    onFocus,
+    placeholder,
+    readOnly,
+    sensitive,
+    showPerChar,
+    title,
+    type,
+    valid,
+    value,
+    warning,
+    ...rest
+  } = props
+  const [focus, setFocus] = useState(false)
+  const [hidden, setHidden] = useState(sensitive)
+  const input = useRef(null)
+  const intl = useIntl()
+
+  const computeByteInputWidth = useCallback(() => {
     const isSafari = isSafariUserAgent()
-
+    const { max } = props
```

Comment:
Deconstruct `max` from the props on line 72, then use it in the callback and add it to the dependency array.

If you extract it like this, you make the hook dependant on the entire prop object.